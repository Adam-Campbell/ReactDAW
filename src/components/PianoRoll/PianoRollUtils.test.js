import {
    createSelectedAndUnselectedNoteArrays,
    getWholeBarsFromString,
    transportPositionStringToSixteenths,
    getTransportLineAttrs,
    createPitchesArray,
    createGridLinesArray,
    isValidNote,
    calculateNoteInfo,
    createTransportBarNumbersArray,
    swapSelectedNoteIds,
    shiftPitchUp,
    shiftPitchDown,
    shiftTimeBackwards,
    shiftTimeForwards,
    findOverlapAlongAxis,
    getNoteIdsForSelectionRange,
    getNoteDurationFromPencilOperation,
    generateNoteObjectForPasting,
    getSortedNoteDataStructs,
    getFirstAvailablePitchInChord
} from './PianoRollUtils';

describe('findOverlapAlongAxis', () => {
    test('correctly identifies an overlap between the elements', () => {
        expect(findOverlapAlongAxis({
           elementALowerBound: 48,
           elementAUpperBound: 632,
           elementBLowerBound: 24,
           elementBUpperBound: 128 
        })).toBe(true);
    });
    test('correctly identifies the absence of an overlap between the elements', () => {
        expect(findOverlapAlongAxis({
            elementALowerBound: 16,
            elementAUpperBound: 128,
            elementBLowerBound: 256,
            elementBUpperBound: 512 
        })).toBe(false);
    });
});