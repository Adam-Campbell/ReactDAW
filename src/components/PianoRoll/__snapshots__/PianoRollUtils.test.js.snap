// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createTransportBarNumbersArray correctly adjusts for different sectionBars values 1`] = `
Array [
  Object {
    "barNumber": 0,
    "xPos": 0,
  },
  Object {
    "barNumber": 1,
    "xPos": 384,
  },
]
`;

exports[`createTransportBarNumbersArray correctly adjusts for different sectionStart values 1`] = `
Array [
  Object {
    "barNumber": 2,
    "xPos": 0,
  },
  Object {
    "barNumber": 3,
    "xPos": 384,
  },
  Object {
    "barNumber": 4,
    "xPos": 768,
  },
  Object {
    "barNumber": 5,
    "xPos": 1152,
  },
]
`;

exports[`createTransportBarNumbersArray correctly creates the transport bar numbers array 1`] = `
Array [
  Object {
    "barNumber": 0,
    "xPos": 0,
  },
  Object {
    "barNumber": 1,
    "xPos": 384,
  },
  Object {
    "barNumber": 2,
    "xPos": 768,
  },
  Object {
    "barNumber": 3,
    "xPos": 1152,
  },
]
`;
